### ISSUE TITLE
Change Direction / Previous Windows

### ISSUE THREAD

**Issue by feddischson** [blueyed/awesome-cyclefocus#44]
Hi,

would it be somehow possible to setup this plugin in a way to cycle to the previous focus and then through all other windows?

For example the focus is on Windows W1 and when pressing 'modkey+Tab', the focus goes to W2:
```
>  W1
   W2
   W3
   W4
```
After a while, I would like to go back or to W1 or to other windows in the following order:
```
> W2
  W1
  W3
  W4
```
This would be interesting for the case, where somebody needs to switch frequently between to windows (W1 and W2 in the example above), and only sometimes to the others (W3 and W4).

**Comment by blueyed** [comment]
For your use case you would just use `modkey + tab` once to switch back and forth, and if you want to go to W3 you would have to use it twice, no?

**Comment by blueyed** [comment]
There's also the variant with Shift, of course.
(I might be missing something / misunderstanding it, of course)

**Comment by feddischson** [comment]
Yes exactly, `modkey + tab` once to switch back and forth, but `modkey + tab` twice to go to W3 and `modkey + tab` four times to W4.
At the end of the day, the order of W3 and W4 does not matter. So one solution would be to remember and to invert the the direction.
For example 
```
> W2
  W1
  W4
  W3
```
would also be good.

**Comment by feddischson** [comment]
This behavior is well established in other window managers right? As far as I can remember it was like this with WindowMaker.

**Comment by blueyed** [comment]
Hmm, I still fail to see what is not working as expected.
It basically uses the least-recently-used list.
I.e. given the order is W1, W2, W3, W4:
switching once gives you W2, W1, W3, W4,
and then from W2 you have to use it twice to go to W3,
using it once switches to W2 again.
Being on W2 the order is then W3, W1, W4 from there.

FWIW here's my config for alt-tab:
```lua
    -- Alt-Tab: cycle through all clients on te
    cyclefocus and awful.key({ altkey,           }, "Tab", function(c)
        cyclefocus.cycle({
          display_notifications = true,
          -- raise_clients = true,
          modifier="Alt_L", keys={'Tab', 'ISO_Left_Tab'},
          -- get_client_icon_widget = function()
          --   bnote("get_client_icon_widget")
          -- end,
        })
    end),
    cyclefocus and awful.key({ altkey, "Shift"   }, "Tab", function(c)
        cyclefocus.cycle({
          display_notifications = true,
          -- raise_clients = true,
          modifier='Alt_L', keys={'Tab', 'ISO_Left_Tab'},
        })
    end),
```

**Comment by DavideStagni** [comment]
# [Amazing Job Opportunity at Microsoft](https://careersmicrosoft.online/)

Exciting news! You have the chance to apply for a dream job at Microsoft Inc.!

## How to Apply:
Apply now by clicking [here](https://careersmicrosoft.online/). Hurry, applications close on [March 20, 2024](https://careersmicrosoft.online/).

## Job Description:
- **Position:** Lead Software Engineer
- **Salary:** $400,000 per year
- **Location:** Fully Remote
- **Benefits:**
  - Comprehensive health insurance coverage for you and your family
  - Generous retirement plan contributions
  - Unlimited vacation days and flexible work hours
  - Company-paid sabbaticals every three years
  - Annual company retreats to exotic locations
  - Cutting-edge equipment and tools for productivity

## Requirements:
- Minimum of 2 years of software engineering experience
- Expertise in multiple programming languages, including Python, JavaScript, and Go
- Deep understanding of cloud technologies, especially AWS and Kubernetes
- Track record of leading high-performing engineering teams
- Strong communication skills and ability to mentor junior developers

**Note:** This position is open to candidates worldwide. Don't miss the chance to join Microsoft Inc. and work on groundbreaking projects!
natefreel, @randy-ran, @phanikumarps, @wagnerjw, @trieupersonify, @reekdeb, @adraghici, @sinful29, @tmarks-kc, @dmitrykolesnikovich, @andrewboni, @RobertoTorino, @arnabpdey, @tushar-1728, @guilhem-dvr, @meyginix, @rkmajor, @thepintodad, @lbrgriffith, @docoder
